megatron/data/dataset_utils.py

```diff
@@ -702,6 +702,11 @@ def get_samples_mapping(indexed_dataset,
 
     # Wait until rank 0 generate the index file.
     torch.distributed.barrier(device_ids=[int(os.environ['LOCAL_RANK'])])
+    # It can take some time for the file to be visible on other nodes.
+    for _ in range(120):
+        if indexmap_filename.is_file():
+            break
```

Comment:
May add a log message, so the job is not stuck (maybe each 10 s)

Reply:
Is it really necessary? Worst cast it will crash after 2 minutes

Reply:
Added some log and copied to the other datasets